CoreInteractiveGraph/test/unit/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/PointSelectionPluginNGTest.java

```diff
@@ -0,0 +1,570 @@
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.graph.interaction.plugins.select;
+
+import au.gov.asd.tac.constellation.graph.GraphElementType;
+import au.gov.asd.tac.constellation.graph.StoreGraph;
+import au.gov.asd.tac.constellation.graph.schema.visual.concept.VisualConcept;
+import au.gov.asd.tac.constellation.plugins.Plugin;
+import au.gov.asd.tac.constellation.plugins.PluginExecution;
+import au.gov.asd.tac.constellation.utilities.graphics.IntArray;
+import static org.testng.Assert.*;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * PointSelectionPlugin Tests.
+ * 
+ * @author sol695510
+ */
+public class PointSelectionPluginNGTest {
+    
+    private StoreGraph storeGraph;
+    
+    // Vertex and transaction attribute IDs respectively.
+    private int vAttrId, tAttrId;
+    
+    // Vertex and transaction IDs respectively.
+    private int vxId1, vxId2, vxId3;
+    private int txId1, txId2, txId3;
+    
+    // Arrays used by the PointSelectionPlugin that contain the selected graph elements during plugin execution.
+    final private IntArray vxIds = new IntArray();
+    final private IntArray txIds = new IntArray();
+    
+    // If true is as if control key is held down so consecutive selections will select multiple elements,
+    // or deselect selected elements.
+    boolean toggleSelection;
+    
+    // If true deleselects all other elements aside from the one being selected.
+    // If toggle selection is true this should be false, otherwise consecutive selections will be deselected.
+    boolean clearSelection;
+    
+    public PointSelectionPluginNGTest() {
+    }
+    
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+    
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+    
+    /**
+     * Creates a basic graph with 3 vertices and 3 transactions with selection attributes added for testing.
+     * 
+     * @throws Exception 
+     */
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+        
+        storeGraph = new StoreGraph();
+        
+        vAttrId = VisualConcept.VertexAttribute.SELECTED.ensure(storeGraph);
+        tAttrId = VisualConcept.TransactionAttribute.SELECTED.ensure(storeGraph);
+        
+        vxId1 = storeGraph.addVertex();
+        vxId2 = storeGraph.addVertex();
+        vxId3 = storeGraph.addVertex();
+        
+        txId1 = storeGraph.addTransaction(vxId1, vxId2, false);
+        txId2 = storeGraph.addTransaction(vxId2, vxId3, false);
+        txId3 = storeGraph.addTransaction(vxId3, vxId1, false);
+        
+        // Ensures arrays don't contain IDs from previous tests when going into a subsequent test.
+        vxIds.clear();
+        txIds.clear();
```

Comment:
if you move the instantiation of vxId and txId here (similar to what you've done with other objects), you should be able to remove these clears (along with any clears at the very end of unit tests)